src/main/java/apoc/graph/Graphs.java

```diff
@@ -137,17 +138,53 @@ else if (data instanceof Iterator) {
 
     @Description("apoc.graph.validateDocument({json}, {config}) yield row - validates the json, return the result of the validation")
     @Procedure(mode = Mode.READ)
-    public Stream<RowResult> validateDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) throws Exception {
-        DocumentToGraph documentToGraph = new DocumentToGraph(db, new GraphsConfig(config));
+    public Stream<RowResult> validateDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+        GraphsConfig graphConfig = new GraphsConfig(config);
+        DocumentToGraph documentToGraph = new DocumentToGraph(db, graphConfig);
         AtomicLong index = new AtomicLong(-1);
-        return getDocumentCollection(document).stream()
+        Collection<Map> documentCollection = getDocumentCollection(document);
+        Set<String> skipLine = new HashSet<>();
+        return documentCollection.stream()
                 .map(elem -> {
+                    long line = index.incrementAndGet();
                     try {
+                        // id, label validation
+                        documentToGraph.prepareData(elem);
                         documentToGraph.validate(elem);
-                        return new RowResult(Util.map("index", index.incrementAndGet(), "message", "Valid"));
+                        Object id = elem.get(graphConfig.getIdField());
```

Comment:
please extract the code in a well named method so it's at the same abstraction as `documentToGraph.prepareData(elem);`